XSLT/transpile.xslt

```diff
@@ -18,36 +19,68 @@
 
   <xsl:include href="modules/functions_module.xslt"/>
 
+  <xsl:param name="atop:pPatternPrefix" as="xs:string">
```

Comment:
Whoa!
I could swear on a ton of bricks I looked before I lept, and it was _not_ handled. My mistake, I guess. In any case, two thoughts:
- I am tossed as to whether it is better to generate the prefix in the main code or in the get-*-pattern-name functions. My instinct is in the functions (as @dmj has it) is better than in the main code (as I have it), but that means we end up repeating the logic for generating a prefix half a dozen times or so. So we should probably factor it out into a separate function.
- [Answering “Not sure why we would do this. If there is no prefix, there is no prefix.”, below.] The short answer is that a pattern prefix is required to avoid name collisions with RELAX NG code sucked in from other namespaces. Of course, there may not _be_ any RELAX NG code sucked in from other namespaces. So seems to me we either need to
1. generate a pattern prefix whether or not the user provides a `@prefix` attribute (which is what the current Stylesheets do, I think); or
1. set up the PLODD schema so that it is invalid to provide a PLODD that has or sucks in  RELAX NG from elsewhere but does _not_ have a `@prefix`; or
1. look to see if any other RELAX NG code is in (or sucked in by) the PLODD, and if so generate a pattern prefix whether or not there is a `@prefix`.

I personally like having the pattern prefixes in the RELAX NG even when they are not strictly necessary, so I lean towards solution (1).